src/test/java/com/aws/greengrass/security/SecurityServiceTest.java

```diff
@@ -345,4 +353,103 @@ void GIVEN_key_and_cert_uri_WHEN_get_mqtt_builder_from_default_THEN_succeed() th
             assertThat(builder, IsNull.notNullValue());
         }
     }
+
+    @Test
+    void GIVEN_key_and_cert_uri_WHEN_getCertificateChain_and_providerImplemented_THEN_getCertChainFromProvider()
+            throws Exception {
+        URI keyUri = new URI("pkcs11:object=key-label");
+        URI certificateUri = new URI("file:///path/to/certificate");
+        when(mockKeyProvider.supportedKeyType()).thenReturn("PKCS11");
+        when(mockKeyProvider.getCertificateChain(keyUri, certificateUri)).thenReturn(Collections.emptyList());
+
+        service.registerCryptoKeyProvider(mockKeyProvider);
+        List<X509Certificate> certs = service.getCertificateChain(keyUri, certificateUri);
+        assertThat(certs, notNullValue());
```

Comment:
Could we also assert that the value we get here is the cert that we expect. Just returning an List with a random `X509Certificate` or an empty array would make this test pass